sound/soc/sof/intel/hda.c

```diff
@@ -992,9 +992,15 @@ static int hda_generic_machine_select(struct snd_sof_dev *sdev)
 				dmic_num = hda_dmic_num;
```

Comment:
I'm actually wondering about this override: do we really want to allow the user specify 4 microphones when the system only has 2 or 1?

Reply:
sorry didn't get it? I understand the dmic_num is coming from nhlt tables, so from bios/manufacturer... we are only filtering out other than 1, 2, 3, 4. Or do you refer to some other place than this patch touches?

Reply:
@juimonen yes, sorry, I meant the `hda_dmic_num` module parameter, which isn't touched by this patch, it just occurred to me while reviewing

Reply:
ok, let's deal with it later if needed, thanks.